activerecord/lib/active_record/connection_adapters/pool_config.rb

```diff
@@ -80,4 +92,5 @@ def discard_pool!
   end
 end
 
+ActiveSupport::ForkTracker.before_fork { ActiveRecord::ConnectionAdapters::PoolConfig.prepare_to_fork! }
```

Comment:
One issue here, is that `after_fork` is simple, because in the child, only the thread that forked survived. 

But `before_fork` is much trickier, because nothing guarantees no thread was spawned, nor that any of these threads isn't using a connection right now.

So we potentially have a big thread safety on our hands here.